File: src/i18n/strings.ts
import type { LanguageCode } from '../types';

const strings = {
  appName: {
    en: 'Theory Prep',
    pt: 'Preparação Teórica',
    fr: 'Prépa Code',
    es: 'Preparación Teórica',
  },
  selectCountryTitle: {
    en: 'Which country are you preparing for?',
    pt: 'Para que país se está a preparar?',
    fr: 'Pour quel pays vous préparez-vous ?',
    es: '¿Para qué país te estás preparando?',
  },
  selectCountrySubtitle: {
    en: 'This determines your questions, road signs, and exam rules.',
    pt: 'Isto determina as suas perguntas, sinais de trânsito e regras de exame.',
    fr: 'Cela détermine vos questions, panneaux et règles d’examen.',
    es: 'Esto determina tus preguntas, señales y normas de examen.',
  },
  selectLanguageTitle: {
    en: 'In which language would you like to study?',
    pt: 'Em que língua gostaria de estudar?',
    fr: 'Dans quelle langue souhaitez-vous étudier ?',
    es: '¿En qué idioma te gustaría estudiar?',
  },
  selectLanguageSubtitle: {
    en: 'You can study in English even if it is not the local language.',
    pt: 'Pode estudar em inglês mesmo que não seja a língua local.',
    fr: 'Vous pouvez étudier en anglais même si ce n’est pas la langue locale.',
    es: 'Puedes estudiar en inglés aunque no sea el idioma local.',
  },
  changeCountryLanguage: {
    en: 'Change country / language',
    pt: 'Mudar país / língua',
    fr: 'Changer de pays / langue',
    es: 'Cambiar país / idioma',
  },
  home: { en: 'Home', pt: 'Início', fr: 'Accueil', es: 'Inicio' },
  practice: { en: 'Practice', pt: 'Praticar', fr: 'Entraînement', es: 'Practicar' },
  exam: { en: 'Exam', pt: 'Exame', fr: 'Examen', es: 'Examen' },
  roadSigns: { en: 'Road Signs', pt: 'Sinais', fr: 'Panneaux', es: 'Señales' },
  mistakes: { en: 'Mistakes', pt: 'Erros', fr: 'Erreurs', es: 'Errores' },
  progress: { en: 'Progress', pt: 'Progresso', fr: 'Progrès', es: 'Progreso' },
  yourProgress: { en: 'Your Progress', pt: 'O seu Progresso', fr: 'Votre Progression', es: 'Tu Progreso' },
  questionsAnswered: {
    en: 'Questions Answered',
    pt: 'Perguntas Respondidas',
    fr: 'Questions Répondues',
    es: 'Preguntas Respondidas',
  },
  accuracy: { en: 'Accuracy', pt: 'Precisão', fr: 'Précision', es: 'Precisión' },
  weakTopic: { en: 'Weakest Topic', pt: 'Tópico Mais Fraco', fr: 'Sujet le Plus Faible', es: 'Tema Más Débil' },
  continueStudying: {
    en: 'Continue Studying',
    pt: 'Continuar a Estudar',
    fr: 'Continuer à Étudier',
    es: 'Seguir Estudiando',
  },
  startExam: { en: 'Start Exam Simulation', pt: 'Iniciar Simulação de Exame', fr: 'Démarrer l’Examen Blanc', es: 'Iniciar Simulacro de Examen' },
  noDataYet: {
    en: 'No practice yet — start your first session!',
    pt: 'Ainda sem prática — comece a sua primeira sessão!',
    fr: 'Pas encore de pratique — commencez votre première session !',
    es: 'Aún sin práctica — ¡empieza tu primera sesión!',
  },
  allCategories: { en: 'All Categories', pt: 'Todas as Categorias', fr: 'Toutes les Catégories', es: 'Todas las Categorías' },
  randomPractice: { en: 'Random Practice', pt: 'Prática Aleatória', fr: 'Entraînement Aléatoire', es: 'Práctica Aleatoria' },
  randomPracticeSubtitle: {
    en: 'A mix of questions from every category',
    pt: 'Uma mistura de perguntas de todas as categorias',
    fr: 'Un mélange de questions de toutes les catégories',
    es: 'Una mezcla de preguntas de todas las categorías',
  },
  practiceByCategory: {
    en: 'Practice by Category',
    pt: 'Praticar por Categoria',
    fr: 'S’entraîner par Catégorie',
    es: 'Practicar por Categoría',
  },
  submitAnswer: { en: 'Submit', pt: 'Confirmar', fr: 'Valider', es: 'Confirmar' },
  continueButton: { en: 'Continue', pt: 'Continuar', fr: 'Continuer', es: 'Continuar' },
  finishButton: { en: 'Finish', pt: 'Terminar', fr: 'Terminer', es: 'Finalizar' },
  correct: { en: 'Correct!', pt: 'Correto!', fr: 'Correct !', es: '¡Correcto!' },
  incorrect: { en: 'Incorrect', pt: 'Incorreto', fr: 'Incorrect', es: 'Incorrecto' },
  correctAnswerWas: {
    en: 'The correct answer is:',
    pt: 'A resposta correta é:',
    fr: 'La bonne réponse est :',
    es: 'La respuesta correcta es:',
  },
  explanation: { en: 'Explanation', pt: 'Explicação', fr: 'Explication', es: 'Explicación' },
  questionOf: { en: 'Question', pt: 'Pergunta', fr: 'Question', es: 'Pregunta' },
  practiceComplete: {
    en: 'Practice Session Complete',
    pt: 'Sessão de Prática Concluída',
    fr: 'Session Terminée',
    es: 'Sesión Completada',
  },
  backToHome: { en: 'Back to Home', pt: 'Voltar ao Início', fr: 'Retour à l’Accueil', es: 'Volver al Inicio' },
  practiceAgain: { en: 'Practice Again', pt: 'Praticar Novamente', fr: 'S’entraîner à Nouveau', es: 'Practicar de Nuevo' },
  timeRemaining: { en: 'Time Remaining', pt: 'Tempo Restante', fr: 'Temps Restant', es: 'Tiempo Restante' },
  examIntroTitle: { en: 'Exam Simulation', pt: 'Simulação de Exame', fr: 'Examen Blanc', es: 'Simulacro de Examen' },
  examIntroDescription: {
    en: 'This simulates the real exam: timed, no immediate feedback, and a pass/fail result at the end.',
    pt: 'Isto simula o exame real: com tempo, sem feedback imediato, e resultado de aprovado/reprovado no final.',
    fr: 'Ceci simule l’examen réel : chronométré, sans retour immédiat, avec un résultat final réussite/échec.',
    es: 'Esto simula el examen real: con tiempo, sin retroalimentación inmediata y resultado de apto/no apto al final.',
  },
  numberOfQuestions: { en: 'Number of Questions', pt: 'Número de Perguntas', fr: 'Nombre de Questions', es: 'Número de Preguntas' },
  timeLimit: { en: 'Time Limit', pt: 'Tempo Limite', fr: 'Durée', es: 'Tiempo Límite' },
  passingScore: { en: 'Passing Score', pt: 'Nota Mínima', fr: 'Score Requis', es: 'Nota Mínima' },
  beginExam: { en: 'Begin Exam', pt: 'Iniciar Exame', fr: 'Commencer l’Examen', es: 'Comenzar Examen' },
  examResults: { en: 'Exam Results', pt: 'Resultados do Exame', fr: 'Résultats de l’Examen', es: 'Resultados del Examen' },
  passed: { en: 'Passed', pt: 'Aprovado', fr: 'Réussi', es: 'Aprobado' },
  failed: { en: 'Failed', pt: 'Reprovado', fr: 'Échoué', es: 'Suspendido' },
  yourScore: { en: 'Your Score', pt: 'A sua Nota', fr: 'Votre Score', es: 'Tu Nota' },
  reviewAnswers: { en: 'Review Answers', pt: 'Rever Respostas', fr: 'Revoir les Réponses', es: 'Revisar Respuestas' },
  retakeExam: { en: 'Retake Exam', pt: 'Repetir Exame', fr: 'Repasser l’Examen', es: 'Repetir Examen' },
  timeUp: { en: 'Time is up!', pt: 'Tempo esgotado!', fr: 'Temps écoulé !', es: '¡Se acabó el tiempo!' },
  noMistakesTitle: {
    en: 'No mistakes yet',
    pt: 'Ainda sem erros',
    fr: 'Aucune erreur pour l’instant',
    es: 'Aún sin errores',
  },
  noMistakesSubtitle: {
    en: 'Questions you answer incorrectly will show up here for focused review.',
    pt: 'As perguntas que responder incorretamente aparecerão aqui para revisão focada.',
    fr: 'Les questions auxquelles vous répondez mal apparaîtront ici pour une révision ciblée.',
    es: 'Las preguntas que respondas mal aparecerán aquí para repasarlas.',
  },
  practiceMistakes: {
    en: 'Practice These Mistakes',
    pt: 'Praticar Estes Erros',
    fr: 'S’entraîner sur ces Erreurs',
    es: 'Practicar Estos Errores',
  },
  incorrectTimes: {
    en: 'answered incorrectly {count}x',
    pt: 'respondida incorretamente {count}x',
    fr: 'répondu incorrectement {count}x',
    es: 'respondida mal {count}x',
  },
  byCategory: { en: 'By Category', pt: 'Por Categoria', fr: 'Par Catégorie', es: 'Por Categoría' },
  recentExams: { en: 'Recent Exam Results', pt: 'Resultados Recentes de Exame', fr: 'Résultats d’Examens Récents', es: 'Resultados Recientes de Examen' },
  noExamsYet: {
    en: 'No exam simulations completed yet.',
    pt: 'Ainda sem simulações de exame concluídas.',
    fr: 'Aucun examen blanc terminé pour l’instant.',
    es: 'Aún no has completado ningún simulacro de examen.',
  },
  loading: { en: 'Loading…', pt: 'A carregar…', fr: 'Chargement…', es: 'Cargando…' },
  meaning: { en: 'Meaning', pt: 'Significado', fr: 'Signification', es: 'Significado' },
} as const;

export type StringKey = keyof typeof strings;

export function getString(lang: LanguageCode, key: StringKey, vars?: Record<string, string | number>): string {
  const entry = strings[key];
  let value: string = entry[lang] ?? entry.en;
  if (vars) {
    for (const [k, v] of Object.entries(vars)) {
      value = value.replace(`{${k}}`, String(v));
    }
  }
  return value;
}

File: TheoryPrep/Localization/Strings.swift
import Foundation

enum StringKey: String, CaseIterable {
    case appName, selectLanguageTitle, selectLanguageSubtitle, selectLanguageActionHint
    case changeLanguage
    case home, today, practice, exam, roadSigns, mistakes, review, progress
    case yourProgress, questionsAnswered, accuracy, weakTopic, continueStudying, startExam, noDataYet
    case allCategories, randomPractice, randomPracticeSubtitle, practiceByCategory
    case submitAnswer, continueButton, finishButton, correct, incorrect, correctAnswerWas, explanation
    case selectOneAnswer, selectAllAnswers, maximumMistakes, mistakeCount
    case questionOf, practiceComplete, backToHome, practiceAgain
    case timeRemaining, examIntroTitle, examIntroDescription, numberOfQuestions, passingScore, beginExam
    case examResults, passed, failed, yourScore, reviewAnswers, retakeExam, timeUp
    case noMistakesTitle, noMistakesSubtitle, practiceMistakes, incorrectTimes
    case byCategory, recentExams, noExamsYet, loading, meaning
    case readinessTitle, readinessNoData, readinessNotReady, readinessReadyOnMeasured
    case readinessSolidLabel, readinessFreshnessLabel, readinessBelowBarLabel, readinessUnmeasuredLabel
    case readinessCeiling, readinessReadyCaveat, readinessNeverStudied, readinessDaysAgoFormat
    case readinessThemesFormat, readinessHowCalculated
    case readinessRuleCoverage, readinessRuleAccuracy, readinessRuleStale, readinessRuleUnmeasurable
    case statusSolid, statusStale, statusNeedsAccuracy, statusBuildingCoverage, statusNotStarted, statusUnmeasurable
    case themeSeenFormat
    case focusWeakSpots, roadSignsLibrary
    case examMetadata, splashSubtitle, signCountFormat, categoryCountFormat
    case opensRoadSignsHint
    case perQuestion, examUntimedBadge, examTimedToggle, examTimedToggleHint
    case confidencePrompt, confidenceSure, confidenceUnsure, confidenceSureHint, confidenceUnsureHint
    case confidentlyWrongTitle, confidentlyWrongBody, confidentBadge
    case knowledgeDrill
    case finishExamPrompt, finishExamConfirm, cancel, unansweredQuestionsFormat
    case exitExamLabel, exitExamTitle, exitExamMessage, exitExamConfirm
}

private typealias L = [LanguageCode: String]

private func entry(_ en: String, _ fr: String) -> L {
    [.en: en, .fr: fr]
}

private func buildStrings() -> [StringKey: L] {
    var s: [StringKey: L] = [:]

    s[.appName] = entry("Theory Prep", "Prépa Code")
    s[.selectLanguageTitle] = entry(
        "In which language would you like to study?",
        "Dans quelle langue souhaitez-vous étudier ?"
    )
    s[.selectLanguageSubtitle] = entry(
        "You can study in English even if it is not the local language.",
        "Vous pouvez étudier en anglais même si ce n’est pas la langue locale."
    )
    s[.selectLanguageActionHint] = entry(
        "Uses this language for questions and explanations.",
        "Utilise cette langue pour les questions et les explications."
    )
    s[.changeLanguage] = entry("Change language", "Changer de langue")
    s[.home] = entry("Home", "Accueil")
    s[.today] = entry("Today", "Aujourd’hui")
    s[.practice] = entry("Practice", "Entraînement")
    s[.exam] = entry("Exam", "Examen")
    s[.roadSigns] = entry("Road Signs", "Panneaux")
    s[.mistakes] = entry("Mistakes", "Erreurs")
    s[.review] = entry("Review", "Révision")
    s[.progress] = entry("Progress", "Progrès")
    s[.yourProgress] = entry("Your Progress", "Votre Progression")
    s[.questionsAnswered] = entry("Questions Answered", "Questions Répondues")
    s[.accuracy] = entry("Accuracy", "Précision")
    s[.weakTopic] = entry("Weakest Topic", "Sujet le Plus Faible")
    s[.continueStudying] = entry("Continue Studying", "Continuer à Étudier")
    s[.startExam] = entry("Start Exam Simulation", "Démarrer l’Examen Blanc")
    s[.noDataYet] = entry(
        "No practice yet — start your first session!",
        "Pas encore de pratique — commencez votre première session !"
    )
    s[.allCategories] = entry("All Categories", "Toutes les Catégories")
    s[.randomPractice] = entry("Random Practice", "Entraînement Aléatoire")
    s[.randomPracticeSubtitle] = entry(
        "All questions from every category, in random order",
        "Toutes les questions de chaque catégorie, dans un ordre aléatoire"
    )
    s[.practiceByCategory] = entry("Practice by Category", "S’entraîner par Catégorie")
    s[.submitAnswer] = entry("Submit", "Valider")
    s[.selectOneAnswer] = entry(
        "Select one answer.",
        "Sélectionnez une réponse."
    )
    s[.selectAllAnswers] = entry(
        "More than one answer is correct.",
        "Plusieurs bonnes réponses."
    )
    s[.maximumMistakes] = entry("Maximum Mistakes", "Fautes Maximales")
    s[.mistakeCount] = entry("mistakes", "fautes")
    s[.continueButton] = entry("Continue", "Continuer")
    s[.finishButton] = entry("Finish", "Terminer")
    s[.correct] = entry("Correct!", "Correct !")
    s[.incorrect] = entry("Incorrect", "Incorrect")
    s[.correctAnswerWas] = entry("The correct answer is:", "La bonne réponse est :")
    s[.explanation] = entry("Explanation", "Explication")
    s[.questionOf] = entry("Question", "Question")
    s[.practiceComplete] = entry("Practice Session Complete", "Session Terminée")
    s[.backToHome] = entry("Back to Home", "Retour à l’Accueil")
    s[.practiceAgain] = entry("Practice Again", "S’entraîner à Nouveau")
    s[.timeRemaining] = entry("Time Remaining", "Temps Restant")
    s[.examIntroTitle] = entry("Exam Simulation", "Examen Blanc")
    s[.examIntroDescription] = entry(
        "Exam-style practice with no correction until the end. Only photo/video questions enter this mock; text-only knowledge drills remain in Practice. The optional timer is a training setting, not an official timing claim.",
        "Entraînement de type examen, sans correction avant la fin. Seules les questions avec photo ou vidéo entrent dans ce test ; les fiches sans média restent dans l’Entraînement. Le chronomètre facultatif est un réglage d’entraînement, pas une durée officielle annoncée."
    )
    s[.numberOfQuestions] = entry("Number of Questions", "Nombre de Questions")
    s[.passingScore] = entry("Passing Score", "Score Requis")
    s[.beginExam] = entry("Begin Exam", "Commencer l’Examen")
    s[.examResults] = entry("Exam Results", "Résultats de l’Examen")
    s[.passed] = entry("Passed", "Réussi")
    s[.failed] = entry("Failed", "Échoué")
    s[.yourScore] = entry("Your Score", "Votre Score")
    s[.reviewAnswers] = entry("Review Answers", "Revoir les Réponses")
    s[.retakeExam] = entry("Retake Exam", "Repasser l’Examen")
    s[.timeUp] = entry("Time is up!", "Temps écoulé !")
    s[.noMistakesTitle] = entry("No mistakes yet", "Aucune erreur pour l’instant")
    s[.noMistakesSubtitle] = entry(
        "Questions you answer incorrectly will show up here for focused review.",
        "Les questions auxquelles vous répondez mal apparaîtront ici pour une révision ciblée."
    )
    s[.practiceMistakes] = entry("Practice These Mistakes", "S’entraîner sur ces Erreurs")
    s[.incorrectTimes] = entry("answered incorrectly %@x", "répondu incorrectement %@x")
    s[.byCategory] = entry("By Category", "Par Catégorie")
    s[.recentExams] = entry("Recent Exam Results", "Résultats d’Examens Récents")
    s[.noExamsYet] = entry("No exam simulations completed yet.", "Aucun examen blanc terminé pour l’instant.")
    s[.loading] = entry("Loading…", "Chargement…")
    s[.meaning] = entry("Meaning", "Signification")
    s[.readinessTitle] = entry("Readiness", "Préparation")
    s[.readinessNoData] = entry("Not measured yet", "Pas encore mesuré")
    s[.readinessNotReady] = entry("Not ready yet", "Pas encore prêt")
    s[.readinessReadyOnMeasured] = entry(
        "Ready on measured themes",
        "Prêt sur les thèmes mesurés"
    )
    s[.readinessSolidLabel] = entry("Themes solid", "Thèmes solides")
    s[.readinessFreshnessLabel] = entry("Last studied", "Dernière révision")
    s[.readinessBelowBarLabel] = entry("Below the bar", "Sous le seuil")
    s[.readinessUnmeasuredLabel] = entry("Not measurable", "Données insuffisantes")
    s[.readinessCeiling] = entry(
        "Bank: %d questions. Past that, this measures how well you remember this app, not how ready you are.",
        "Banque : %d questions. Au-delà, cela mesure votre mémoire de l’app, pas votre préparation."
    )
    s[.readinessReadyCaveat] = entry(
        "%d themes still cannot be measured with the current question bank.",
        "%d thèmes restent impossibles à mesurer avec la banque actuelle."
    )
    s[.readinessNeverStudied] = entry("Never", "Jamais")
    s[.readinessDaysAgoFormat] = entry("%d days ago", "il y a %d j")
    s[.readinessThemesFormat] = entry("%d of %d", "%d sur %d")
    s[.readinessHowCalculated] = entry("How this is calculated", "Comment c’est calculé")
    s[.readinessRuleCoverage] = entry(
        "You have seen at least %d%% of the theme’s questions",
        "Vous avez vu au moins %d %% des questions du thème"
    )
    s[.readinessRuleAccuracy] = entry(
        "At least %d%% correct on that theme",
        "Au moins %d %% de bonnes réponses sur ce thème"
    )
    s[.readinessRuleStale] = entry(
        "Revised within the last %d days",
        "Révisé il y a moins de %d jours"
    )
    s[.readinessRuleUnmeasurable] = entry(
        "Themes with fewer than %d questions are not scored at all",
        "Les thèmes de moins de %d questions ne sont pas notés du tout"
    )
    s[.statusSolid] = entry("Solid", "Solide")
    s[.statusStale] = entry("Needs refreshing", "À rafraîchir")
    s[.statusNeedsAccuracy] = entry("Accuracy too low", "Précision insuffisante")
    s[.statusBuildingCoverage] = entry("Too few questions seen", "Trop peu de questions vues")
    s[.statusNotStarted] = entry("Not started", "Non commencé")
    s[.statusUnmeasurable] = entry("Not enough questions", "Données insuffisantes")
    s[.themeSeenFormat] = entry("%d/%d seen", "%d/%d vues")
    s[.focusWeakSpots] = entry("Focus on weak spots", "Révisez vos points faibles")
    s[.roadSignsLibrary] = entry("French road signs library", "Bibliothèque des panneaux français")
    s[.examMetadata] = entry(
        "%d questions · %d correct to pass · independent practice",
        "%d questions · %d bonnes réponses · entraînement indépendant"
    )
    s[.perQuestion] = entry("Practice timer", "Chrono d’entraînement")
    s[.confidencePrompt] = entry("Validate your answer:", "Validez votre réponse :")
    s[.confidenceSure] = entry("I'm sure", "Je sais")
    s[.confidenceUnsure] = entry("Not sure", "J\u{2019}hésite")
    s[.confidenceSureHint] = entry(
        "Submits your answer and records that you were sure.",
        "Valide votre réponse et indique que vous étiez sûr de la connaître."
    )
    s[.confidenceUnsureHint] = entry(
        "Submits your answer and records that you were guessing.",
        "Valide votre réponse et indique que vous hésitiez."
    )
    s[.confidentlyWrongTitle] = entry("%d answered wrongly while sure", "%d erreurs alors que vous pensiez savoir")
    s[.confidentlyWrongBody] = entry(
        "These matter most: you thought you knew, and the answer was wrong. Nothing prompts you to revise them, so they are the likeliest to cost a mark.",
        "Ce sont les plus importantes : vous pensiez savoir, et la réponse était fausse. Rien ne vous pousse à les réviser, ce sont donc celles qui risquent le plus de vous coûter un point."
    )
    s[.confidentBadge] = entry("You thought you knew", "Vous pensiez savoir")
    s[.knowledgeDrill] = entry("Knowledge drill", "Fiche de connaissances")
    s[.examUntimedBadge] = entry("Untimed", "Sans chronomètre")
    s[.examTimedToggle] = entry(
        "%d-second practice timer per question",
        "Chronomètre d’entraînement de %d secondes par question"
    )
    s[.examTimedToggleHint] = entry(
        "Turn this off if you need more time. Everything else stays the same, and your result is recorded either way.",
        "Désactivez-le s\u{2019}il vous faut plus de temps. Le reste ne change pas, et votre résultat est enregistré dans tous les cas."
    )
    s[.splashSubtitle] = entry("French driving theory", "Code de la route — France")
    s[.signCountFormat] = entry("%d signs", "%d panneaux")
    s[.categoryCountFormat] = entry("%d categories", "%d catégories")
    s[.opensRoadSignsHint] = entry(
        "Opens the road signs reference.",
        "Ouvre le guide des panneaux routiers."
    )
    s[.finishExamPrompt] = entry("Finish this exam?", "Terminer cet examen ?")
    s[.finishExamConfirm] = entry("Finish Exam", "Terminer l’examen")
    s[.cancel] = entry("Keep Working", "Continuer l’examen")
    s[.exitExamLabel] = entry("Leave exam", "Quitter l\u{2019}examen")
    s[.exitExamTitle] = entry("Leave this exam?", "Quitter cet examen ?")
    s[.exitExamMessage] = entry(
        "Nothing will be saved: no result, and no mistakes added to your review list.",
        "Rien ne sera enregistré : ni résultat, ni erreurs ajoutées à vos révisions."
    )
    s[.exitExamConfirm] = entry("Leave", "Quitter")
    s[.unansweredQuestionsFormat] = entry(
        "%d unanswered questions will be marked incorrect.",
        "%d questions sans réponse seront comptées comme incorrectes."
    )

    return s
}

private let strings: [StringKey: L] = {
    let values = buildStrings()
    precondition(
        Set(values.keys) == Set(StringKey.allCases),
        "Every StringKey must have an English and French localization entry"
    )
    precondition(
        values.values.allSatisfy { $0[.en] != nil && $0[.fr] != nil },
        "Every localized string must provide both English and French"
    )
    return values
}()

func localizedString(_ key: StringKey, _ language: LanguageCode?, _ args: CVarArg...) -> String {
    localizedString(key, language, arguments: args)
}

func localizedString(_ key: StringKey, _ language: LanguageCode?, arguments: [CVarArg]) -> String {
    let lang: LanguageCode = language ?? .en
    let template: String = strings[key]?[lang] ?? strings[key]?[.en] ?? key.rawValue
    return arguments.isEmpty ? template : String(format: template, arguments: arguments)
}
